parsec/arena.c

```diff
@@ -176,6 +176,62 @@ parsec_arena_release_chunk(parsec_arena_t* arena,
     arena->data_free(chunk);
 }
 
+int  parsec_arena_allocate_copy_device_private(parsec_arena_t *arena,
```

Comment:
I do not understand that function? This is intended to allocate memory on the device_private, yet we allocate a single arena_chunk from the `arena->data_malloc`, then we fill-it from the CPU. Either 1) this is allocating CPU memory or 2) this doesn't work with UVM; 3) this cannot and should not be used to allocate GPU device memory?

Reply:
1) yes. The code of the GPU assumes (e.g. if a successor task cannot execute on a GPU but receives this data copy as input) that all data_t have a version on device 0 (CPU). So, when we allocate a data copy on demand on the GPU, we call his function to be sure the CPU version will be reserved;
2) UVM is not our current model, for these data we manage the explicit transfers
3) we use the normal mechanism to reserve GPU memory when we have a NEW data on GPU, as we would for any data_copy, but we just need to instantiate the CPU version for the rest of the code to work

Reply:
Always CPU memory. The name `device_private` might be misleading, it has nothing to do with a hardware `device`.

Reply:
OK, I understand the confusion now. Will change the name of the function.

Reply:
I don't think it is confusing for people working in that level of the runtime, because it mirror the name of the field in the data copy structure.

Reply:
Merge the two arena allocation functions to factorize code, and have the data copy inherit the device index of the arena.

Reply:
Done... @abouteiller please check if your requested changes have been done?

Reply:
deduplication is done as planned :+1:. 

I made a new comment about the function name, let's take a last look at it and then move on https://github.com/ICLDisco/parsec/pull/318#discussion_r917049565